src/commonTest/kotlin/kotools/types/collection/NotEmptyMapTest.kt

```diff
@@ -1,15 +1,77 @@
 package kotools.types.collection
 
-import kotlinx.serialization.*
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerializationException
 import kotlinx.serialization.builtins.MapSerializer
 import kotlinx.serialization.builtins.serializer
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
 import kotlinx.serialization.json.Json
 import kotools.types.contentShouldEqual
 import kotools.types.shouldHaveAMessage
 import kotlin.random.Random
 import kotlin.test.*
 
 class NotEmptyMapTest {
+    @Test
+    fun notEmptyMapOf_should_pass() {
+        val head: Pair<String, Int> = "a" to Random.nextInt()
+        val tail: Array<Pair<String, Int>> = arrayOf(
+            "b" to Random.nextInt(),
+            "c" to Random.nextInt()
+        )
+        assertEquals(
+            actual = notEmptyMapOf(head, *tail).toMap(),
+            expected = mapOf(head, *tail)
+        )
+    }
+
+    @Test
+    fun map_toNotEmptyMap_should_pass_with_a_not_empty_Map() {
+        val map: Map<Char, Int> = mapOf('a' to Random.nextInt())
+        val result: Result<NotEmptyMap<Char, Int>> = map.toNotEmptyMap()
+        result.getOrThrow()
+            .toMap()
+            .entries contentShouldEqual map.entries
+    }
+
+    @Test
+    fun map_toNotEmptyMap_should_fail_with_an_empty_Map() {
+        val map: Map<Char, Int> = emptyMap()
+        val result: Result<NotEmptyMap<Char, Int>> = map.toNotEmptyMap()
+        val error: IllegalArgumentException =
+            assertFailsWith(block = result::getOrThrow)
+        error.shouldHaveAMessage()
+    }
```

Comment:
Refactor this test by introducing a `shouldFailWithIllegalArgumentException` function in the [`kotools.types.Assertions.kt`](https://github.com/kotools/types/blob/5d1dc8f98f2220e96081381809cde9d261d04f8f/src/commonTest/kotlin/kotools/types/Assertions.kt) file.

```kotlin
// In the kotools.types.Assertions.kt file.
internal inline fun <T> T.shouldFailWithIllegalArgumentException(
    block: T.() -> Unit
): IllegalArgumentException = TODO()
```

```suggestion
    @Test
    fun map_toNotEmptyMap_should_fail_with_an_empty_Map() {
        val map: Map<Char, Int> = emptyMap()
        val result: Result<NotEmptyMap<Char, Int>> = map.toNotEmptyMap()
        result.shouldFailWithIllegalArgumentException { getOrThrow() }
            .shouldHaveAMessage()
    }
```